src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -264,6 +273,32 @@ impl<S: StateStore> StorageTableInner<S, EitherSerde> {
             .collect();
         let pk_serializer = OrderedRowSerde::new(pk_data_types, order_types);
 
+        let partial_row_deserializer = if versioned {
+            let value_output_indices_dedup = value_output_indices
+                .iter()
+                .sorted()
+                .dedup()
+                .copied()
+                .collect::<Vec<_>>();
+            let output_row_in_value_output_indices_dedup = value_output_indices
+                .iter()
+                .map(|&di| {
+                    value_output_indices_dedup
+                        .iter()
+                        .position(|&pi| di == pi)
+                        .unwrap()
+                })
+                .collect_vec();
+            let partial_row_serde = ColumnAwareSerde::new(
```

Comment:
For `row_serde`, we use different Serde implementation depending on whether `versioned` is true. Do we need to do the same trick here?

Reply:
It is already. For !versioned, partial_row_deserializer is not available.
`let partial_row_deserializer = if versioned`

Reply:
Oh my bad. I missed that if check at the beginning. Thanks for the clarification.